fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkimport/ImportPartitionMapper.java

```diff
@@ -130,128 +137,137 @@
+            result = cosClient.listObjectsV2(request);
+            for (S3ObjectSummary objectSummary : result.getObjectSummaries()) {
+                boolean isToBeProccessed = false;
+                if (dsDataLocationInfo != null && !dsDataLocationInfo.trim().isEmpty()) {
+                    if (objectSummary.getKey().startsWith(dsDataLocationInfo.trim())) {
+                        isToBeProccessed = true;
+                    }
+                } else {
+                    isToBeProccessed = true;
+                }
+                if (isToBeProccessed) {
+                    logger.info("getFhirDataSourcesForObjectStore: ObjectStorge Object(" + objectSummary.getKey()
+                            + ") - " + objectSummary.getSize() + " bytes.");
+                    if (objectSummary.getSize() > 0) {
+                        fhirDataSources.add(new FhirDataSource(dsTypeInfo, objectSummary.getKey()));
+                    }
+                }
+            }
+        } while (result != null && result.isTruncated());
+
+        return fhirDataSources;
+    }
+
+    private List<FhirDataSource> getFhirDataSources(JsonArray dataSourceArray, BulkImportDataSourceStorageType type)
+            throws Exception {
+        List<FhirDataSource> fhirDataSources = new ArrayList<>();
+        for (JsonValue jsonValue : dataSourceArray) {
+            JsonObject dataSourceInfo = jsonValue.asJsonObject();
+            String dsTypeInfo = dataSourceInfo.getString(Constants.IMPORT_INPUT_RESOURCE_TYPE);
+            String dsDataLocationInfo = dataSourceInfo.getString(Constants.IMPORT_INPUT_RESOURCE_URL);
+
+            switch (type) {
+            case HTTPS:
+            case FILE:
+                fhirDataSources.add(new FhirDataSource(dsTypeInfo, dsDataLocationInfo));
+                break;
+            case AWSS3:
+            case IBMCOS:
+                fhirDataSources.addAll(getFhirDataSourcesForObjectStore(dsTypeInfo, dsDataLocationInfo));
+                break;
+            default:
+                break;
+            }
+        }
+
+        return fhirDataSources;
+    }
 
     @Override
     public PartitionPlan mapPartitions() throws Exception {
-        JsonReader reader = Json.createReader(new StringReader(
-                new String(Base64.getDecoder().decode(dataSourcesInfo), StandardCharsets.UTF_8)));
+        JsonReader reader =
+                Json.createReader(new StringReader(
+                        new String(Base64.getDecoder().decode(dataSourcesInfo), StandardCharsets.UTF_8)));
         JsonArray dataSourceArray = reader.readArray();
         reader.close();
 
-        List<FhirDataSource> fhirDataSources = getFhirDataSources(dataSourceArray, BulkImportDataSourceStorageType.from(dataSourceStorageType));
-
+        BulkImportDataSourceStorageType dsType = BulkImportDataSourceStorageType.from(dataSourceStorageType);
+        List<FhirDataSource> fhirDataSources = getFhirDataSources(dataSourceArray, dsType);
```

Comment:
because dsType is only used once, so, not really sure if it's good to use a local variable for it.

Reply:
I split them as I was debugging and needed to log out.  I can move it back